libs/utils/src/compressed_tallies.ts

```diff
@@ -124,19 +144,56 @@ function getContestTalliesForCompressedContest(
   }
 }
 
+export function decodeCompressedTally(
+  encodedCompressedTally: string,
+  election: Election
+): CompressedTally {
+  const buffer = Buffer.from(encodedCompressedTally, 'base64url');
+  const uint16Array = new Uint16Array(
+    buffer.buffer,
+    buffer.byteOffset,
+    buffer.byteLength / 2
+  );
+  const compressedTally: CompressedTally = [];
+  let offset = 0;
+  for (const contest of election.contests) {
+    if (contest.type === 'yesno') {
+      compressedTally.push(
+        Array.from(
+          uint16Array.slice(offset, offset + 5)
```

Comment:
This and the other `slice` results are completely unchecked and may not return a subarray of the expected length:

```ts
> (new Uint16Array(2)).slice(0, 5)
Uint16Array(2) [ 0, 0 ]
```
I believe we should be able to calculate the expected length upfront and reject it if it doesn't match, then these operations in the loop should be safe.

Also, we can make the `5` less of a magic number and force the build to fail if it gets out of sync by using something like this:

```ts
  const yesNoContestCompressedTallyLength: YesNoContestCompressedTally['length'] = 5;
```

We can do something similar for candidate contests, though it'd be slightly more complicated.